src/main/java/jenkins/plugins/slack/ActiveNotifier.java

```diff
@@ -91,12 +91,12 @@ private void notifyStart(AbstractBuild build, String message) {
         if (lastBuild != null) {
             AbstractBuild<?, ?> previousBuild = lastBuild.getPreviousCompletedBuild();
             if (previousBuild == null) {
-                getSlack(build).publish(message, "good");
+                getSlack(build).publish(message, "good", listener);
```

Comment:
what do you think about instead logging it where the other log statements go here?
https://github.com/jenkinsci/slack-plugin/blob/227150df87b8aea207150d283c51ce0a95d21b16/src/main/java/jenkins/plugins/slack/SlackNotifier.java#L360

I don't think passing a listener down to the publisher is great as its mixing two concepts, the listener has nothing to do with it

Reply:
@timja I agree. I fixed that, and it's much cleaner now. I've looked at other plugins on github and they seem to put that kind of logging in the `perform()` method.